### PR TITLE
Des 777

### PR DESCRIPTION
**Pull request by wyseguyonline**
Phone number

### CODE REVIEW

File: demo/demoPages/components/PhoneNumberPage.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      <div className="code">
+        <h3>Props</h3>
+        <h4>Optional:</h4>
+          <ul>
+            <li className="li-props">labelText:String === {`<PhoneNumber labelText="Your label" />`}</li>
+            <li className="li-props">Text value displayed above the Input</li>
+            <li className="li-props">infoMessage:String === {`<PhoneNumber infoMessage="How informative!" />`}</li>
+            <li className="li-props">Message displayed below the Input</li>
+            <li className="li-props">errorMessage:String === {`<PhoneNumber errorMessage="Error message" />`}</li>
+            <li className="li-props">Mesage displayed below the Input during an error state.</li>
```
Comment (StommePoes) [inline demo/demoPages/components/PhoneNumberPage.js]: "mesage"

File: demo/demoPages/components/PhoneNumberPage.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            <li className="li-props">onChange:Function === {`<PhoneNumber onChange={(n) => console.log(n)}`}</li>
+            <li className="li-props">This handler is called each time the {`<input />`} changes its textual value.</li>
+            <li className="li-props">value:String ==={`<PhoneNumber value="Populate the input's value" />`}</li>
+            <li className="li-props">onKeyDown:Function === {`<PhoneNumber onKeyDown={(key) => console.log(key)}`}</li>
+            <li className="li-props">Can be used in special cases, for example when enter is pressed.</li>
+            <li className="li-props">disabled:Boolean === {`<PhoneNumber disabled />`}</li>
+            <li className="li-props">Disables both the {`<input />`} and {`<select />`}.</li>
+            <li className="li-props">onCountryChange:Function</li>
+            <li className="li-props">Callback called when the selected country changes.</li>
+            <li className="li-props">countries:Array === {`<Calendar countries=['US', 'UK'] />`}</li>
```
Comment (StommePoes) [inline demo/demoPages/components/PhoneNumberPage.js]: why is this calendar?

File: src/components/PhoneNumber/PhoneNumber.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          onCountryChange={onCountryChange}
+          countries={countries}
+          international={international}
+          convertToNational={convertToNational}
+          selectMaxItems={selectMaxItems}
+          className={className}
+          inputClassName={errorInput}
+          metadata={metadata}
+        />
+        {infoMessage && <p className="pe-input--info_message">{infoMessage}</p>}
```
Comment (StommePoes) [inline src/components/PhoneNumber/PhoneNumber.js]: Can these info and error messages please follow the input versions where these messages are programmatically associated with the inputs using aria-describedby + id?

File: src/components/PhoneNumber/component/rrui.css [hunk 4]
```diff
@@ -0,0 +1,366 @@
+/*===========================================================
+= Input                                                     =
+===========================================================*/
+
+.rrui__input
+{
+	/* `position: relative` is for the floating label absolute positioning
```
Comment (StommePoes) [inline src/components/PhoneNumber/component/rrui.css]: looks like we're not using floating labels, but are using placeholders and labels.

File: src/components/PhoneNumber/resources/flags/xk.svg [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" width="800" height="600">
+	<title>Flag of Kosovo</title>
```
Comment (StommePoes) [inline src/components/PhoneNumber/resources/flags/xk.svg]: If we're going to be using svg flags this component comes with, that's okay, but either we adjust this svg to use the titles accessibly (in a way supported by more browsers) *or* we set aria-hidden=true so there's not some people getting "flag of kosovo +383" and other people getting "+383" with no country name.

Also to stop IE from making the SVG itself a tab stop, focusable=false should be added (to the group of whatever SVGs we use)

File: src/components/PhoneNumber/source/Input.js [hunk 6]
```diff
@@ -0,0 +1,1235 @@
...
+		// Custom "International" phone number type icon.
+		internationalIcon : PropTypes.element.isRequired,
+
+		// Should the initially passed phone number `value`
+		// be converted to a national phone number for its country.
+		// (is `false` by default)
+		convertToNational : PropTypes.bool.isRequired,
+
+		// HTML `tabindex` attribute for the country select
+		selectTabIndex    : PropTypes.number,
```
Comment (StommePoes) [inline src/components/PhoneNumber/source/Input.js]: I'm wondering why this is an attribute for these components. Developers might actually use this attribute. That would be bad.

File: src/components/PhoneNumber/source/Input.js [hunk 7]
```diff
@@ -0,0 +1,1235 @@
...
+
+		// Defines the height of the dropdown country select list
+		selectMaxItems    : PropTypes.number,
+
+		// (deprecated, use just `tabIndex` instead)
+		// HTML `tabindex` attribute for the phone number `<input/>`
+		inputTabIndex     : PropTypes.number,
```
Comment (StommePoes) [inline src/components/PhoneNumber/source/Input.js]: same, why would we ever put a tabindex attr on an input?

File: src/components/PhoneNumber/source/Input.js [hunk 8]
```diff
@@ -0,0 +1,1235 @@
...
+
+		// `aria-label` for the `<Select/>`'s `<button/>`
+		selectAriaLabel : PropTypes.string,
+
+		// `aria-label` for the `<Select/>`'s "Close" button
```
Comment (StommePoes) [inline src/components/PhoneNumber/source/Input.js]: A close button on a select is weeeeeeeeird. Weird weird weird.

File: src/components/PhoneNumber/source/Input.js [hunk 9]
```diff
@@ -0,0 +1,1235 @@
...
+				// might be comfortable with entering an international phone number
+				// (i.e. with country code) rather than the local one.
+				// Therefore such possibility is given.
+				//
+				return character
+			}
+		}
+		// For digits.
+		// Converts wide-ascii and arabic-indic numerals to European numerals.
+		// E.g. in Iraq they don't write `+442323234` but rather `+٤٤٢٣٢٣٢٣٤`.
```
Comment (StommePoes) [inline src/components/PhoneNumber/source/Input.js]: interesting that numbers are apparently written left to write but everything else right to left? Or they didn't want to deal with bidi tags and placing the cursor....

File: src/components/PhoneNumber/source/Input.js [hunk 10]
```diff
@@ -0,0 +1,1235 @@
...
+			{
+				return true
+			}
+		}
+	}
+
+	// Can be called externally
+	focus = () =>
+	{
+		ReactDOM.findDOMNode(this.input).focus()
```
Comment (StommePoes) [inline src/components/PhoneNumber/source/Input.js]: do we have any convention here, as in would we rather use refs instead? How does findDOMNode affect possible future isomorphic rendering?

File: src/components/PhoneNumber/source/Input.js [hunk 11]
```diff
@@ -0,0 +1,1235 @@
...
+							style={ inputStyle }
+							metadata={ metadata }
+							className='pe-textInput'
+							/><span className='pe-input_underline'></span></div>
+					}
+				</div>
+
+				{/* Error message */}
+				{ error && indicateInvalid &&
+					<div className={ classNames('rrui__input-error', 'react-phone-number-input__error') }>
```
Comment (StommePoes) [inline src/components/PhoneNumber/source/Input.js]: needs to follow the a11y standards similar to regular inputs, aria-describedby+id

File: src/components/PhoneNumber/source/Input.js (hunk 6 above)
Comment (wyseguyonline) [reply src/components/PhoneNumber/source/Input.js]: This isn't exposed on the main component.

File: src/components/PhoneNumber/source/Input.js (hunk 10 above)
Comment (mhomolak) [reply src/components/PhoneNumber/source/Input.js]: It's very unlikely we go pure isomorphic on everything so no convention currently.

**Comment by StommePoes** [comment]
Since this is a custom dropdown it probably needs some of the stuff from custom dropdown... I'll play with it

**Pull request closed without merging**